save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt

```diff
@@ -0,0 +1,274 @@
+/**
+ * A view for registration
+ */
+
+@file:Suppress("FILE_WILDCARD_IMPORTS", "WildcardImport")
+
+package com.saveourtool.save.frontend.components.views
+
+import com.saveourtool.save.domain.ImageInfo
+
+import com.saveourtool.save.frontend.components.inputform.InputTypes
+import com.saveourtool.save.frontend.components.inputform.inputTextFormRequired
+import com.saveourtool.save.frontend.http.getUser
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.classLoadingHandler
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.utils.AvatarType
+import com.saveourtool.save.v1
+import com.saveourtool.save.validation.FrontendRoutes
+import com.saveourtool.save.validation.isValidName
+
+import csstype.ClassName
+import org.w3c.dom.HTMLInputElement
+import org.w3c.dom.asList
+import org.w3c.fetch.Headers
+import org.w3c.xhr.FormData
+import react.*
+import react.dom.aria.ariaLabel
+import react.dom.events.ChangeEvent
+import react.dom.html.ButtonType
+import react.dom.html.InputType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.form
+import react.dom.html.ReactHTML.h1
+import react.dom.html.ReactHTML.img
+import react.dom.html.ReactHTML.input
+import react.dom.html.ReactHTML.label
+import react.dom.html.ReactHTML.main
+import react.dom.html.ReactHTML.span
+
+import kotlinx.browser.window
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * `Props` retrieved from router
+ */
+external interface RegistrationProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+}
+
+/**
+ * [State] of registration view component
+ */
+external interface RegistrationViewState : State {
+    /**
+     * Conflict error message
+     */
+    var conflictErrorMessage: String?
+
+    /**
+     * Flag to handle uploading a file
+     */
+    var isUploading: Boolean
+
+    /**
+     * Image to owner avatar
+     */
+    var image: ImageInfo?
+
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
```

Comment:
Why do we need a copy of UserInfo in component's state? Can we use value from props?

Reply:
Removed it